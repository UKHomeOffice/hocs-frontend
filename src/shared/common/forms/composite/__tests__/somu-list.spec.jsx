

import React from 'react';
import WrappedSomuList from '../somu-list.jsx';
import { MemoryRouter } from 'react-router-dom';

describe('Somu list component', () => {

    const PAGE = { params: { caseId: '1234', stageId: '5678' } };
    const NAME = 'somu_list';
    const ITEM_NAME = 'item';
    const BASE_URL = 'http://localhost:8080';
    const MOCK_CALLBACK = jest.fn();

    const DEFAULT_PROPS = {
        page: PAGE,
        name: NAME,
        itemName: ITEM_NAME,
        baseUrl: BASE_URL,
        updateState: MOCK_CALLBACK
    };

    beforeEach(() => {
        MOCK_CALLBACK.mockReset();
    });

    it('should render with default props', () => {
        const OUTER = shallow(<WrappedSomuList {...DEFAULT_PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(<SomuList />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with label when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            label: 'Test somu list'
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(<SomuList />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with error when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            error: 'Validation has failed'
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(<SomuList />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with additional when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            className: 'my-css-class'
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(<SomuList />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with item link passed in props', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestTeam' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            itemLinks: [
                {
                    'action': 'remove',
                    'label': 'Remove'
                }
            ]
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with MpamTable renderer', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { table: 'MpamTable' } },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestTeam' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with default render when empty renderer added', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { } },
            active: true };
        const somuItems = [{ uuid: 'test', data: { uuid: 'test', contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestTeam' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render when CompTable renderer added', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { table: 'CompTable' } },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with default render when renderers object doesn\'t exist', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestUnit' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with choices when table renderer exists', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { table: 'MpamTable' } },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestTeam' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            choices: [ {
                'value': 'TestTeam',
                'label': 'Test Team'
            }]
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with add link when passed in props', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { table: 'MpamTable' } },
            active: true };
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            primaryLink: {
                'action': 'add',
                'label': 'Add'
            }
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should execute callback on initialization', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        mount(<SomuList />);
        expect(MOCK_CALLBACK).toHaveBeenCalledTimes(1);
        expect(MOCK_CALLBACK).toHaveBeenCalledWith({ [DEFAULT_PROPS.name]: '[]' });
    });

    it('should include hideSidebar query param on primary and item links', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { } },
            active: true };
        const somuItems = [{ uuid: 'test', data: { contributionBusinessArea: 'TestBusinessArea', contributionBusinessUnit: 'TestTeam' }, deleted: false }];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            hideSidebar: true,
            itemLinks: [
                {
                    'action': 'remove',
                    'label': 'Remove'
                }
            ],
            primaryLink: {
                'action': 'add',
                'label': 'Add'
            }
        };

        const OUTER = shallow(<WrappedSomuList { ...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render status column correctly', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: { renderers: { table: 'MpamTable' } },
            active: true };
        const somuItems = [
            { uuid: 'test',
                data: {
                    contributionBusinessArea: 'TestBusinessArea',
                    contributionBusinessUnit: 'TestTeam',
                    contributionStatus: 'contributionCancelled'
                },
                deleted: false },
            { uuid: 'test',
                data: {
                    contributionBusinessArea: 'TestBusinessArea',
                    contributionBusinessUnit: 'TestTeam',
                    contributionStatus: 'contributionReceived'
                },
                deleted: false },
            { uuid: 'test',
                data: {
                    contributionBusinessArea: 'TestBusinessArea',
                    contributionBusinessUnit: 'TestTeam',
                    contributionDueDate: '2020-01-01'
                },
                deleted: false },
            { uuid: 'test',
                data: {
                    contributionBusinessArea: 'TestBusinessArea',
                    contributionBusinessUnit: 'TestTeam',
                    contributionDueDate:  '2050-01-01'
                },
                deleted: false },
            { uuid: 'test',
                data: {
                    contributionBusinessArea: 'TestBusinessArea',
                    contributionBusinessUnit: 'TestTeam',
                    contributionDueDate: 'xx'
                },
                deleted: false }
        ];
        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            hideSidebar: true,
            itemLinks: [
                {
                    'action': 'remove',
                    'label': 'Remove'
                }
            ],
            primaryLink: {
                'action': 'add',
                'label': 'Add'
            }
        };

        const OUTER = shallow(<WrappedSomuList { ...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render ApprovalTypeTable with Due', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: {
                renderers: {
                    table: 'ApprovalRequestTable'
                }
            },
            active: true };

        const somuItems = [{
            uuid: 'test',
            data: {
                approvalRequestDueDate: '2021-09-01',
                approvalRequestCreatedDate: '2021-08-31',
                approvalRequestForBusinessUnit: 'TestTeam'
            },
            deleted: false
        }];

        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            choices: [ {
                'value': 'TestTeam',
                'label': 'Test Team'
            }]
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render ApprovalTypeTable with 1 Complete - approved, 1 Cancelled, 1 Complete - rejected', () => {
        const somuType = { uuid: 'test',
            caseType: 'tesCaseType',
            type: 'testType',
            schema: {
                renderers: {
                    table: 'ApprovalRequestTable'
                }
            },
            active: true };

        const somuItems = [{
            uuid: 'test - 2',
            data: {
                approvalRequestStatus: 'approvalRequestCancelled',
                approvalRequestDueDate: '2021-09-01',
                approvalRequestCreatedDate: '2021-08-31',
                approvalRequestForBusinessUnit: 'TestTeam2',
                approvalRequestCancellationNote: 'Test note'
            },
            deleted: false
        }, {
            uuid: 'test - 3',
            data: {
                approvalRequestStatus: 'approvalRequestResponseReceived',
                approvalRequestDueDate: '2021-09-01',
                approvalRequestDecision: 'approved',
                approvalRequestResponseBy: 'Something',
                approvalRequestCreatedDate: '2021-08-31',
                approvalRequestResponseNote: 'dcadcadf',
                approvalRequestForBusinessUnit: 'TestTeam3',
                approvalRequestResponseReceivedDate: '2021-08-31'
            },
            deleted: false
        },{
            uuid: 'test - 4',
            data: {
                approvalRequestStatus: 'approvalRequestResponseReceived',
                approvalRequestDueDate: '2021-09-01',
                approvalRequestDecision: 'rejected',
                approvalRequestResponseBy: 'Something',
                approvalRequestCreatedDate: '2021-08-31',
                approvalRequestResponseNote: 'dcadcadf',
                approvalRequestForBusinessUnit: 'TestTeam4',
                approvalRequestResponseReceivedDate: '2021-08-31'
            },
            deleted: false
        }];

        const PROPS = {
            ...DEFAULT_PROPS,
            somuType,
            somuItems,
            choices: [ {
                'value': 'TestTeam',
                'label': 'Test Team'
            }, {
                'value': 'TestTeam2',
                'label': 'Test Team 2'
            }, {
                'value': 'TestTeam3',
                'label': 'Test Team 3'
            }, {
                'value': 'TestTeam4',
                'label': 'Test Team 4'
            }]
        };
        const OUTER = shallow(<WrappedSomuList {...PROPS} />);
        const SomuList = OUTER.props().children;
        const WRAPPER = render(
            <MemoryRouter>
                <SomuList page={PAGE} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

});
