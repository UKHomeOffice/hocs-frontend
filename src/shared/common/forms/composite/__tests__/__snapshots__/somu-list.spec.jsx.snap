// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Somu list component should include hideSidebar query param on primary and item links 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              class="govuk-label"
            >
              test
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <p>
      <a
        class="govuk-body govuk-link"
        href="/case/1234/stage/5678/somu/test/testType/tesCaseType/add?hideSidebar=true"
      >
        Add
      </a>
    </p>
  </fieldset>
</div>
`;

exports[`Somu list component should render ApprovalTypeTable with 1 Complete - approved, 1 Cancelled, 1 Complete - rejected 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="Cancelled approval request: Test Team 2"
              class="govuk-label"
              title="Cancelled approval request: Test Team 2"
            >
              Test Team 2
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Cancelled"
              class="govuk-label"
              title="Cancelled"
            >
              Cancelled
            </label>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="Test Team 3"
              class="govuk-label"
              title="Test Team 3"
            >
              Test Team 3
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Complete - approved"
              class="govuk-label"
              title="Complete - approved"
            >
              Complete - approved
            </label>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="Test Team 4"
              class="govuk-label"
              title="Test Team 4"
            >
              Test Team 4
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Complete - rejected"
              class="govuk-label"
              title="Complete - rejected"
            >
              Complete - rejected
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render ApprovalTypeTable with Due 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="Test Team"
              class="govuk-label"
              title="Test Team"
            >
              Test Team
            </label>
          </td>
          <td
            class="govuk-table__cell date-warning"
          >
            <label
              aria-label="Overdue 01/09/2021"
              class="govuk-label"
              title="Overdue 01/09/2021"
            >
              Overdue 01/09/2021
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render status column correctly 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="Cancelled contribution request: TestBusinessArea - TestTeam"
              class="govuk-label"
              title="Cancelled contribution request: TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Cancelled"
              class="govuk-label"
              title="Cancelled"
            >
              Cancelled
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - TestTeam"
              class="govuk-label"
              title="TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Complete"
              class="govuk-label"
              title="Complete"
            >
              Complete
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - TestTeam"
              class="govuk-label"
              title="TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell date-warning"
          >
            <label
              aria-label="Overdue 01/01/2020"
              class="govuk-label"
              title="Overdue 01/01/2020"
            >
              Overdue 01/01/2020
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - TestTeam"
              class="govuk-label"
              title="TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Due 01/01/2050"
              class="govuk-label"
              title="Due 01/01/2050"
            >
              Due 01/01/2050
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - TestTeam"
              class="govuk-label"
              title="TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Due null"
              class="govuk-label"
              title="Due null"
            >
              Due null
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=true"
            >
              Remove
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <p>
      <a
        class="govuk-body govuk-link"
        href="/case/1234/stage/5678/somu/test/testType/tesCaseType/add?hideSidebar=true"
      >
        Add
      </a>
    </p>
  </fieldset>
</div>
`;

exports[`Somu list component should render when CompTable renderer added 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea"
              class="govuk-label"
              title="TestBusinessArea"
            >
              TestBusinessArea
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Due null"
              class="govuk-label"
              title="Due null"
            >
              Due null
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with MpamTable renderer 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - TestTeam"
              class="govuk-label"
              title="TestBusinessArea - TestTeam"
            >
              TestBusinessArea - TestTeam
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Due null"
              class="govuk-label"
              title="Due null"
            >
              Due null
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with add link when passed in props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
    <p>
      <a
        class="govuk-body govuk-link"
        href="/case/1234/stage/5678/somu/test/testType/tesCaseType/add?hideSidebar=false"
      >
        Add
      </a>
    </p>
  </fieldset>
</div>
`;

exports[`Somu list component should render with additional when passed in props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset my-css-class"
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with choices when table renderer exists 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              aria-label="TestBusinessArea - Test Team"
              class="govuk-label"
              title="TestBusinessArea - Test Team"
            >
              TestBusinessArea - Test Team
            </label>
          </td>
          <td
            class="govuk-table__cell "
          >
            <label
              aria-label="Due null"
              class="govuk-label"
              title="Due null"
            >
              Due null
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with default props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with default render when empty renderer added 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              class="govuk-label"
            >
              test
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with default render when renderers object doesn't exist 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              class="govuk-label"
            >
              test
            </label>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with error when passed in props 1`] = `
<div
  class="govuk-form-group govuk-form-group--error"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <span
      class="govuk-error-message"
      id="somu_list-error"
    >
      Validation has failed
    </span>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with item link passed in props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <label
              class="govuk-label"
            >
              test
            </label>
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              class="govuk-link"
              href="/case/1234/stage/5678/somu/test/testType/tesCaseType/item/test/remove?hideSidebar=false"
            >
              Remove
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Somu list component should render with label when passed in props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="somu_list"
  >
    <legend
      class="govuk-fieldset__legend"
      id="somu_list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        Test somu list
      </span>
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
  </fieldset>
</div>
`;
